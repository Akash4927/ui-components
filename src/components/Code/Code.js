import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { isFunction, isString, trim, noop } from 'lodash';

const CopyNotice = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  padding: 10px 15px;
  border-radius: ${props => props.theme.borderRadius.soft};
  background-color: ${props => props.theme.colors.purple800};
  transition: opacity 300ms ease;
  opacity: 0;
  font-size: ${props => props.theme.fontSizes.small};
  color: ${props => props.theme.colors.purple50};
`;

const CodeWrapper = styled.div`
  position: relative;
  padding: 10px 15px;
  background-color: ${props => props.theme.colors.purple800};
  box-sizing: border-box;
  border-radius: ${props => props.theme.borderRadius.soft};
  cursor: pointer;

  &:hover ${CopyNotice} {
    opacity: 0.9;
  }
`;

const Content = styled.div`
  flex-grow: 1;
  overflow: auto;
  color: ${props => props.theme.colors.purple50};
`;

const Pre = styled.pre`
  margin: 10px 0;
  font-family: ${props => props.theme.fontFamilies.monospace};
  font-size: ${props => props.theme.fontSizes.small};
`;

const trimString = node => (isString(node) ? trim(node) : node);

/**
 * Code allows for easy rendering of code snippets which can easliy be copied to
 * the clipboard by clicking the element or selecting a portion of the code
 */
class Code extends Component {
  static displayName = 'Code';

  state = {
    isCopying: false,
  };

  componentDidUpdate = () => {
    // reselect text after re-render
    if (this.state.selectedRange) {
      const prevRange = this.state.selectedRange;
      const sel = window.getSelection();
      sel.removeAllRanges();
      sel.addRange(prevRange);
      // clear it for the next interaction
      this.setState({ selectedRange: null });
    }
  };

  handleCopyClick = () => {
    // Create a textarea element in which the text of the code will be copied.
    // Set the value of the textarea, select it to set it as the active element,
    // then copy the value of that element to the clipboard.
    // It's 2017 and this is still the best way to copy text on button click...
    if (isFunction(this.props.onCopy)) {
      this.props.onCopy();
    }
    const selection = document.getSelection();
    const selectionString = selection.toString();
    // if user has selected text, save that to state so it can be reselected
    if (selectionString !== '') {
      this.setState({
        selectedRange: selection.getRangeAt(0),
      });
    }
    // console.dir(document.getSelection().anchorNode);
    const code =
      selectionString === '' ? this.preNode.textContent : selectionString;
    const txtArea = document.createElement('textarea');
    // Safari doesn't allow for assigning an object literal to `style`.
    txtArea.class = 'hidden-textarea';
    // Make sure the code ends in a newline to execute all the commands in multiline code.
    if (code.charAt(code.length - 1) === '\n') {
      txtArea.value = code;
    } else {
      txtArea.value = `${code}\n`;
    }
    document.body.appendChild(txtArea);

    txtArea.select();

    try {
      document.execCommand('copy');
    } catch (e) {
      throw e;
    }
    document.body.removeChild(txtArea);

    // show the Copied to clipboard notice temporarily
    this.setState({ isCopying: true });
    setTimeout(() => {
      this.setState({ isCopying: false });
    }, 3000);
  };

  render() {
    const { children } = this.props;
    const { isCopying } = this.state;

    const copyText = isCopying ? 'Copied to clipboard' : 'Copy to clipboard';
    const copyIcon = isCopying ? 'files-o' : 'check';

    return (
      <CodeWrapper onClick={this.handleCopyClick}>
        <Content>
          <Pre innerRef={e => (this.preNode = e)}>
            {isFunction(children) ? children() : trimString(children)}
          </Pre>
        </Content>

        <CopyNotice>
          <i className={`fa fa-${copyIcon}`} /> {copyText}
        </CopyNotice>
      </CodeWrapper>
    );
  }
}

Code.propTypes = {
  // Children can be anything that React can render
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,

  // onCopy will be called when the CodeWrapper is clicked
  onCopy: PropTypes.func,
};

Code.defaultProps = {
  onCopy: noop,
};

export default Code;
