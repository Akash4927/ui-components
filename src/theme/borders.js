import { kebabCase, forEach } from 'lodash';

export const borderRadius = {
  none: '0',
  soft: '2px',
  circle: '50%',
};

// Collects all theme border radii as SCSS vars.
export function themeBorderRadiiAsScss() {
  const themeBorderRadii = [];

  forEach(borderRadius, (value, name) => {
    themeBorderRadii.push(`$border-radius-${kebabCase(name)}: ${value}`);
  });

  return themeBorderRadii;
}
