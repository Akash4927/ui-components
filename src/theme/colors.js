import { isString, kebabCase, forEach } from 'lodash';

export const colors = {
  // uncategorized colors
  aliceBlue: '#f0f8ff',
  amethystSmoke: '#9292b7',
  athens: '#e2e2ec',
  athensGray: '#eeeef4',
  black: '#1a1a1a',
  blueHaze: '#c0c0d5',
  comet: '#5b5b88',
  cometDark: '#5a5a7b',
  doveGray: '#737373',
  eastBay: '#4a5d87',
  gallery: '#f4f4f4',
  gunpowder: '#3c3c5b',
  kimberly: '#7d7da8',
  kimberlyDark: '#656597',
  logan: '#b1b1cb',
  mischka: '#dfdfea',
  mulledWine: '#46466a',
  mystic: '#dae4ea',
  silver: '#ccc',
  turquoise: '#037aa9',
  white: '#fff',
  whiteLilac: '#f0f0fa',
  whiteSmoke: '#fafafc',
  promQL: {
    /**
     * GHColors theme by Avi Aryan (http://aviaryan.in)
     * Inspired by Github syntax coloring
     */
    comment: '#bbbbbb',
    string: '#e3116c',
    punctuation: '#393a34',
    entity: '#36acaa',
    metricName: '#2aa198',
    attrName: '#00a4db',
    function: '#dc322f',
    deleted: '#9a050f',
    tag: '#00009f',
    // Dropdown colors,
    salmon: '#ff7c7c',
  },
  // use these for new stuff!
  primary: {
    charcoal: '#32324b',
    lavender: '#8383ac',
  },
  accent: {
    blue: '#00d2ff',
    orange: '#ff4b19',
  },
  status: {
    // Use these when colorizing elements that should indicate a state or outcome, ie Alerts.
    success: '#38bd93', // green
    error: '#c7254e', // red
    warning: '#d4ab27', // amber
    info: '#049cd7', // blue
    disabled: '#9a9a9a',
  },
  // Used by PrometheusGraph component
  graphThemes: {
    blue: [
      '#c7e9b4',
      '#7ecdbb',
      '#1eb5eb',
      '#1d91bf',
      '#235fa9',
      '#253393',
      '#084181',
    ],
    purple: [
      '#c1d4e7',
      '#9fbddb',
      '#8d95c6',
      '#8282ab',
      '#89429e',
      '#800f7a',
      '#0b0533',
    ],
    mixed: [
      '#c7e9b4',
      '#c1d4e7',
      '#7ecdbb',
      '#9fbddb',
      '#1eb5eb',
      '#8d95c6',
      '#1d91bf',
      '#8282ab',
      '#235fa9',
      '#89429e',
      '#253393',
      '#800f7a',
      '#084181',
      '#0b0533',
    ],
  },
  // Third-party specific colors - not to be used in the theme!
  thirdParty: {
    // Google single-click login
    cornflowerBlue: '#4285f4',
    azure: '#3769bb',
  },
};

// Flattens and collects all theme colors as SCSS vars
export function themeColorsAsScss() {
  const ignoreKeys = ['graphThemes'];
  const themeColors = [];

  forEach(colors, (value, name) => {
    const colorPrefix = `$color-${kebabCase(name)}`;
    if (ignoreKeys.includes(name)) return;

    if (isString(value)) {
      themeColors.push(`${colorPrefix}: ${value}`);
    } else {
      forEach(value, (innerValue, subname) => {
        if (ignoreKeys.includes(subname)) return;
        themeColors.push(`${colorPrefix}-${kebabCase(subname)}: ${innerValue}`);
      });
    }
  });

  return themeColors;
}
