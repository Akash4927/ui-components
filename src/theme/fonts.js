import { kebabCase, forEach } from 'lodash';

export const fontFamilies = {
  regular: "'proxima-nova', Helvetica, Arial, sans-serif",
  monospace: "'Roboto Mono', monospace",
};

export const fontSizes = {
  huge: '48px',
  extraLarge: '32px',
  large: '22px',
  normal: '16px',
  small: '14px',
  tiny: '12px',
};

// Collects all theme font vars as SCSS vars.
export function themeFontsAsScss() {
  const themeFonts = [];

  forEach(fontFamilies, (value, name) => {
    themeFonts.push(`$font-family-${kebabCase(name)}: ${value}`);
  });

  forEach(fontSizes, (value, name) => {
    themeFonts.push(`$font-size-${kebabCase(name)}: ${value}`);
  });

  return themeFonts;
}
