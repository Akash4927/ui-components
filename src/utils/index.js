import intersection from 'lodash/intersection';
import first from 'lodash/first';
import keys from 'lodash/keys';

// Return the value of a set of props against a lookup object.
// Helps when using "ternary" props, ie: <Text small bold />
// Third argument is the default value to return in the event of an `undefined`;
export const lookupValue = (props, lookup, _default) => {
  const matches = intersection(keys(props), keys(lookup));

  if (matches.length > 1) {
    throw new Error(`You have specified conflicting props: ${JSON.stringify(matches)}`);
  }

  return lookup[first(matches)] || _default;
};
