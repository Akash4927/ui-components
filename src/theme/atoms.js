import { darken } from 'polished';

import { colors } from './colors';

// component-specific
export const atoms = {
  Alert: {
    default: {
      color: colors.black,
      background: colors.gallery,
      borderColor: darken(0.15, colors.gallery),
    },
    info: {
      color: colors.white,
      background: colors.status.info,
      borderColor: darken(0.15, colors.status.info),
    },
    success: {
      color: colors.white,
      background: colors.status.success,
      borderColor: darken(0.15, colors.status.success),
    },
    warning: {
      color: colors.white,
      background: colors.status.warning,
      borderColor: darken(0.15, colors.status.warning),
    },
    error: {
      color: colors.white,
      background: colors.status.error,
      borderColor: darken(0.15, colors.status.error),
    },
  },
  Button: {
    default: {
      color: colors.black,
      background: colors.white,
      hoverColor: colors.purple800,
      hoverBackground: darken(0.05, colors.white),
    },
    primary: {
      color: colors.white,
      background: colors.turquoise,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.turquoise),
    },
    danger: {
      color: colors.white,
      background: colors.status.error,
      hoverColor: colors.white,
      hoverBackground: darken(0.05, colors.status.error),
    },
    disabled: {
      color: colors.doveGray,
      background: colors.gallery,
      hoverColor: colors.doveGray,
      hoverBackground: colors.gallery,
    },
  },
};
