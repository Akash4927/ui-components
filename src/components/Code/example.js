import React from 'react';
import { trim } from 'lodash';
import styled from 'styled-components';

import { Example, Info } from '../../utils/example';

import { Input } from '..';

import Code from '.';

const Highlight = styled.span`
  color: ${props => props.theme.colors.blue400};
`;

export default class CodeExample extends React.Component {
  state = {
    value: '<ACCESS_KEY>',
  };

  onCopy = () => {
    console.log('Copied'); // eslint-disable-line no-console
  };

  onChange = e => {
    this.setState({
      value: e.target.value,
    });
  };

  render() {
    return (
      <div>
        <Input placeholder="Dynamic content" onChange={this.onChange} />

        <Info>Long single line command</Info>
        <Example>
          <Code onCopy={this.onCopy}>
            kubectl create secret generic -n weave cloudwatch
            --from-literal=access-key-id=xyz
            --from-literal=secret-access-key=abc
          </Code>
        </Example>

        <Info>Multiline file</Info>
        <Example>
          <Code isFile onCopy={this.onCopy}>
            {`
{
  "AccessKey": {
    "AccessKeyId": "<YOUR_ACCESS_TOKEN>",
    "SecretAccessKey": "<YOUR_SECRET_ACCESS_KEY>"
  }
}
          `}
          </Code>
        </Example>

        <Info>Multiline single command</Info>
        <Example>
          <Code onCopy={this.onCopy}>
            {`
kubectl create secret generic -n weave cloudwatch \\
  --from-literal=access-key-id=xyz \\
  --from-literal=secret-access-key=abc
          `}
          </Code>
        </Example>

        <Info>Multiline multi command string</Info>
        <Example>
          <Code onCopy={this.onCopy} multiCommand>
            {`
sudo curl -L git.io/scope -o /usr/local/bin/scope
sudo chmod a+x /usr/local/bin/scope
scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
          `}
          </Code>
        </Example>

        <Info>Multiline multi command JSX</Info>
        <Example>
          <Code onCopy={this.onCopy} multiCommand>
            <div>sudo curl -L git.io/scope -o /usr/local/bin/scope</div>
            <div>sudo chmod a+x /usr/local/bin/scope</div>
            <div>
              scope launch <wbr />
              https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
            </div>
          </Code>
        </Example>

        <Info>Multiline multi command function</Info>
        <Example>
          <Code onCopy={this.onCopy} multiCommand>
            {() => `
sudo curl -L git.io/scope -o /usr/local/bin/scope
sudo chmod a+x /usr/local/bin/scope
scope launch https://${this.state.value}
          `}
          </Code>
        </Example>

        <Info>Dynamic content in template string</Info>
        <Example>
          <Code isFile onCopy={this.onCopy}>
            {() =>
              trim(`
{
  "AccessKey": {
    "AccessKeyId": "${this.state.value}",
    "SecretAccessKey": "<YOUR_SECRET_ACCESS_KEY>"
  }
}
          `)
            }
          </Code>
        </Example>

        <Info>Dynamic content using template strings and JSX</Info>
        <Example>
          <Code onCopy={this.onCopy}>
            <span>
              {'kubectl create secret generic -n weave cloudwatch \\'}
              {'\n  '}
              <strong>--from-literal=access-key-id=</strong>
              <Highlight>{this.state.value}</Highlight>
              {' \\\n  '}
              {'--from-literal=secret-access-key='}
            </span>
          </Code>
        </Example>

        <Info>Dynamic content using HTML</Info>
        <Example>
          <Code onCopy={this.onCopy}>
            <div>
              kubectl create secret generic -n weave cloudwatch \<br />
              <span>
                <strong>{'  '}--from-literal=access-key-id=</strong>
                {this.state.value} \<br />
              </span>
              <span>{'  '}--from-literal=secret-access-key=</span>
            </div>
          </Code>
        </Example>

        <Info>Dynamic content using HTML</Info>
        <Example>
          <Code onCopy={this.onCopy}>
            <div>
              <div>{'{'}</div>
              <div>
                {'  '}&quot;AccessKey&quot;: {'{'}
              </div>
              <div>
                {'    '}
                <strong>&quot;AccessKeyId&quot;</strong>: &quot;
                {this.state.value}&quot;,
              </div>
              <div>
                {'    '}
                <strong>&quot;SecretAccessKey&quot;</strong>:{' '}
                {'"<YOUR_SECRET_ACCESS_KEY>"'}
              </div>
              <div>
                {'  '}
                {'}'}
              </div>
              <div>{'}'}</div>
            </div>
          </Code>
        </Example>
      </div>
    );
  }
}
