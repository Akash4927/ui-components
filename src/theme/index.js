import { colors, themeColorsAsScss } from './colors';
import { fontSizes, fontFamilies, themeFontsAsScss } from './fonts';
import { borderRadius, themeBorderRadiiAsScss } from './borders';
import { layers, themeLayersAsScss } from './layers';
import { spacing, themeSpacingsAsScss } from './spacings';
import { overlayIconSize, textColor, themeMiscVarsAsScss } from './misc';
import { boxShadow } from './box-shadows';
import { atoms } from './atoms';

export function themeVarsAsScss() {
  const themeVariables = []
    .concat(themeColorsAsScss())
    .concat(themeLayersAsScss())
    .concat(themeFontsAsScss())
    .concat(themeBorderRadiiAsScss())
    .concat(themeSpacingsAsScss())
    .concat(themeMiscVarsAsScss());
  return `${themeVariables.join('; ')};`;
}

const theme = {
  // Component-specific
  atoms,

  // Box shadows
  boxShadow,

  // Borders
  borderRadius,

  // Colors
  colors,

  // Fonts
  fontFamilies,
  fontSizes,

  // z-index layers
  layers,

  // Spacings
  spacing,

  // Misc
  overlayIconSize,
  textColor,
};

export default theme;
