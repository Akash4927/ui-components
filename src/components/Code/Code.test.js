import React from 'react';
import enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { withTheme } from '../../utils/theme';
import Code from '.';

enzyme.configure({ adapter: new Adapter() });

document.execCommand = () => {};
document.getSelection = () => ({
  toString: () => '',
});

describe('<Code />', () => {
  let props;
  let wrapper;
  let onCopy;

  beforeEach(() => {
    onCopy = jest.fn();
    props = {
      onCopy,
    };
    wrapper = shallow(
      withTheme(<Code {...props}>such code much programming</Code>)
    );
  });

  it('should call onCopy when the button is clicked', () => {
    wrapper = mount(
      withTheme(<Code {...props}>such code much programming</Code>)
    );
    expect(onCopy).not.toHaveBeenCalled();
    wrapper.find('Code__CopyNotice').simulate('click');
    expect(onCopy).toHaveBeenCalledTimes(1);
  });

  it('should render correctly', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render multi line string correctly with multiCommand=true', () => {
    wrapper = shallow(
      withTheme(
        <Code multiCommand>{`
sudo curl -l git.io/scope -o /usr/local/bin/scope
sudo chmod a+x /usr/local/bin/scope
scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
      `}</Code>
      )
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render multi line function correctly with multiCommand=true', () => {
    wrapper = shallow(
      withTheme(
        <Code multiCommand>
          {() => `
sudo curl -l git.io/scope -o /usr/local/bin/scope
sudo chmod a+x /usr/local/bin/scope
scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
      `}
        </Code>
      )
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render multi line jsx correctly with multiCommand=true', () => {
    wrapper = shallow(
      withTheme(
        <Code multiCommand>
          <div>sudo curl -L git.io/scope -o /usr/local/bin/scope</div>
          <div>sudo chmod a+x /usr/local/bin/scope</div>
          <div>
            scope launch <wbr />
            https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
          </div>
        </Code>
      )
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
