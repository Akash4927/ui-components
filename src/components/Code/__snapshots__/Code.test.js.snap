// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> should render correctly 1`] = `
<Code__CodeWrapper
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Code__ScrollWrap>
    <Code__Content>
      <Code__Pre
        innerRef={[Function]}
      >
        such code much programming
      </Code__Pre>
    </Code__Content>
  </Code__ScrollWrap>
  <Code__CopyNotice
    isCopying={false}
    isHovered={false}
  >
    Copy to clipboard
  </Code__CopyNotice>
</Code__CodeWrapper>
`;

exports[`<Code /> should render multi line function correctly with multiCommand=true 1`] = `
<Code__CodeWrapper
  onClick={[Function]}
>
  <Code__Content>
    <Code__Pre
      innerRef={[Function]}
    >
      <Code__PaddedLine
        key="0"
      >
        sudo curl -l git.io/scope -o /usr/local/bin/scope
        
        
      </Code__PaddedLine>
      <Code__PaddedLine
        key="1"
      >
        sudo chmod a+x /usr/local/bin/scope
        
        
      </Code__PaddedLine>
      <Code__PaddedLine
        key="2"
      >
        scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
        
        
      </Code__PaddedLine>
    </Code__Pre>
  </Code__Content>
  <Code__CopyNotice>
    <i
      className="fa fa-check"
    />
     
    Copy to clipboard
  </Code__CopyNotice>
</Code__CodeWrapper>
`;

exports[`<Code /> should render multi line jsx correctly with multiCommand=true 1`] = `
<Code__CodeWrapper
  onClick={[Function]}
>
  <Code__Content>
    <Code__Pre
      innerRef={[Function]}
    >
      <Code__PaddedLine
        key="0/.0"
      >
        <div>
          sudo curl -L git.io/scope -o /usr/local/bin/scope
        </div>
        
        
      </Code__PaddedLine>
      <Code__PaddedLine
        key="1/.1"
      >
        <div>
          sudo chmod a+x /usr/local/bin/scope
        </div>
        
        
      </Code__PaddedLine>
      <Code__PaddedLine
        key="2/.2"
      >
        <div>
          scope launch 
          <wbr />
          https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
        </div>
        
        
      </Code__PaddedLine>
    </Code__Pre>
  </Code__Content>
  <Code__CopyNotice>
    <i
      className="fa fa-check"
    />
     
    Copy to clipboard
  </Code__CopyNotice>
</Code__CodeWrapper>
`;

exports[`<Code /> should render multi line string correctly with multiCommand=true 1`] = `
<Code__CodeWrapper
  onClick={[Function]}
>
  <Code__Content>
    <Code__Pre
      innerRef={[Function]}
    >
      <Code__PaddedLine
        key="0"
      >
        sudo curl -l git.io/scope -o /usr/local/bin/scope
        
        
      </Code__PaddedLine>
      <Code__PaddedLine
        key="1"
      >
        sudo chmod a+x /usr/local/bin/scope
        
        
      </Code__PaddedLine>
      <Code__PaddedLine
        key="2"
      >
        scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
        
        
      </Code__PaddedLine>
    </Code__Pre>
  </Code__Content>
  <Code__CopyNotice>
    <i
      className="fa fa-check"
    />
     
    Copy to clipboard
  </Code__CopyNotice>
</Code__CodeWrapper>
`;
